infrastructure/src/main/java/org/corfudb/infrastructure/CompactorLeaderServices.java

```diff
@@ -0,0 +1,497 @@
+            }
+            if (readCache.containsKey(table)) {
+                LivenessMetadata previousStatus = readCache.get(table);
+                if (previousStatus.getStreamTail() < currentStreamTail) {
+                    readCache.put(table, new LivenessMetadata(previousStatus.getHeartbeat(), currentStreamTail, currentTime));
+                } else if (previousStatus.getHeartbeat() < syncHeartBeat) {
+                    readCache.put(table, new LivenessMetadata(syncHeartBeat, previousStatus.getStreamTail(), currentTime));
+                } else if (previousStatus.getStreamTail() == currentStreamTail || previousStatus.getHeartbeat() == syncHeartBeat) {
+                    if (currentTime - previousStatus.getTime() > timeout && !handleSlowCheckpointers(table)) {
+                        readCache.clear();
+                        return;
+                    }
+                }
+            } else {
+                readCache.put(table, new LivenessMetadata(syncHeartBeat, currentStreamTail, currentTime));
+            }
+        }
+    }
+
+    private int getIdleCount() {
+        int idleCount = 0;
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            idleCount = txn.executeQuery(checkpointingStatusTable, entry -> (
+                    entry.getPayload()).getStatus() == StatusType.IDLE).size();
+            syslog.trace("Number of idle tables: {}", idleCount);
+            txn.commit();
+        }
+        return idleCount;
+    }
+
+    private void handleNoActiveCheckpointers(long timeout, long currentTime) {
+
+        //Find the number of tables with IDLE status
+        long idleCount = getIdleCount();
+        if (LivenessValidatorHelper.getPrevActiveTime() < 0 || idleCount < LivenessValidatorHelper.getPrevIdleCount()) {
+            syslog.trace("Checkpointing in progress...");
+            LivenessValidatorHelper.updateValues(idleCount, currentTime);
+            return;
+        }
+        CheckpointingStatus managerStatus = null;
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            managerStatus = (CheckpointingStatus) txn.getRecord(
+                    DistributedCompactor.COMPACTION_MANAGER_TABLE_NAME,
+                    DistributedCompactor.COMPACTION_MANAGER_KEY).getPayload();
+            txn.commit();
+        } catch (Exception e) {
+            syslog.warn("Unable to acquire Manager Status");
+        }
+        if (idleCount == 0 || currentTime - LivenessValidatorHelper.getPrevActiveTime() > timeout &&
+                managerStatus != null && managerStatus.getStatus() == StatusType.STARTED_ALL) {
+            readCache.clear();
+            LivenessValidatorHelper.clear();
+            finishCompactionCycle();
+        } else if (currentTime - LivenessValidatorHelper.getPrevActiveTime() > timeout &&
+                managerStatus != null && managerStatus.getStatus() == StatusType.STARTED) {
```

Comment:
please get rid of null

Reply:
If the managerStatus is never set, it can only be null. It's not possible to set it with a value at startup as it might affect the existing flow of the coordination logic.

Reply:
The managerStatus is not of type enum. It's of type CheckpointStatus. 
managerStatus.getStatus() returns enum which is always set if managerStatus != null